src/other/utils.jl

```diff
@@ -477,3 +477,16 @@ function _findall(B::BitVector)::Union{UnitRange{Int}, Vector{Int}}
     end
     @assert false "should not be reached"
 end
+
+# Creates a vector like `vect` without the missing type and copies only data at indices in `inds`
+# Fuses two operations that were previously done by getindex + disallowmissing
+# unsafe because it does not check for `inds` values to be not missing (ensured by the caller)
+# Called by `_getindex_disallowmissing(df,...)`
+function _getindex_disallowmissing(vect::AbstractVector{T}, inds::AbstractVector{<:Integer}) where {T}
+    output = Vector{nonmissingtype(T)}(undef, length(inds))
```

Comment:
`vect` does not have to be `Vector`. Commonly it will be `PooledVector` or `CategoricalVector`. Typically we use `Tables.allocatecolumn` to get such behavior.

Additionally we might special case "pooled" vectors to preserve their pools as this will be faster (but this is a wider issue so it can be skipped in this PR).

Reply:
Excellent point. I wasn't aware of `Tables.allocatecolumn`, so thanks for the tip!

The penalty is incredibly high but at least it's general
```
vect=rand(1:10, 10^7) |>allowmissing
@btime output = Vector{nonmissingtype(eltype($vect))}(undef, 10^7);
# 31.397 μs (2 allocations: 38.15 MiB)
@btime Tables.allocatecolumn(eltype($vect),10^7);
# 2.007 ms (2 allocations: 47.68 MiB)

cvect=CategoricalArray(vect)
@btime output = Vector{nonmissingtype(eltype($vect))}(undef, 10^7);
# 20.340 μs (2 allocations: 76.29 MiB)
@btime Tables.allocatecolumn(eltype($vect),10^7);
# 3.684 ms (2 allocations: 85.83 MiB)
```

Reply:
I've had a quick look if I could add a special case for pooled vectors, but I believe the best way is to actually provide a general fallback that will leverage each special type's own methods.

Quick benchmark: general fallback is much faster (I presume that the element-by-element access on overall type is too expensive.

```
function general_fallback(vect,inds)
    vect[inds]|>disallowmissing
end

vect=rand(1:10, 10^7) |>allowmissing
cvect=CategoricalArray(vect)
row_mask = rand(10^7) .<= 0.5
row_inds = DataFrames._findall(row_mask)

@btime x=_getindex_disallowmissing($cvect, $row_inds);
# 33.966 ms (5000926 allocations: 171.69 MiB)

@btime x=general_fallback($cvect, $row_inds);
# 6.407 ms (20 allocations: 38.16 MiB)

# and we don't seem to pay a penalty for using indices instead of a bitvector mask
@btime x=general_fallback($cvect, $row_mask);
# 8.980 ms (20 allocations: 38.16 MiB)
```

If we wanted to specialize on pooled vectors, I'd probably need some pointers
PooledArray
- it's a dependency, so we could specialize directly
- It's probably not a good idea to use internals (copy pool, getindex refs vector and update the rest)
- it might be possible to fallback to general getindex|>disallowmissing fairly easily

CategoricalArrays
- I couldn't find any trait or reference with which I could specialize
- I went through DataAPI but the best is default array which we have already via Tables.allocatecolumn
- Alternatively, I could add weakdeps (1.9 extensions functionality) and fallback with Requires, but that's probably too involved for such narrow use case